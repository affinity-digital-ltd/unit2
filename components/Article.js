import { Component } from 'react'
import moment from 'moment'

import CommentForm from '../components/CommentForm'

import Auth from '../lib/Auth'

class Article extends Component {
  constructor (props) {
    super(props)
    this.login = this.login.bind(this)
    this.auth = new Auth()
  }

  canComment () {
    if (this.auth.isAuthenticated()) {
      return <CommentForm postComment={this.props.postComment} />
    } else {
      return <div className='card'>
        <div className='card-body'>
          <p>If you would like to leave a comment, then please login</p>
          <a className='btn btn-success btn-lg' onClick={this.login} >Login</a>
        </div>
      </div>
    }
  }

  login () {
    this.auth.login()
  }

  render () {
    const { post } = this.props

    return <div className='row'>
      <div className='col-lg-8 offset-lg-2'>
        <h1 className='c-article__header'>{post.title}</h1>
        <hr className='c-article__rule' />
        <p className='c-article__intro' >{post.intro}</p>
        <p className='c-article__body'>{post.body}</p>

        <div className='c-article__comments'>
          {post.comments.map((comment, index) => {
            return <div className='card c-article__comment' key={index}>
              <div className='card-header'>
                <time dateTime={comment.created_at}>{moment(comment.created_at).format('LL')}</time>
              </div>
              <div className='card-body'>
                <h6 className='card-subtitle mb-2 text-muted'>
                  {comment.name}
                </h6>
                {comment.body}
              </div>
            </div>
          })}
        </div>
        {this.canComment()}
      </div>
    </div>
  }
}

export default Article
