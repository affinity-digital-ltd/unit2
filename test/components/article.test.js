import Article from '../../components/Article'

describe('<Article />', () => {
  test('it renders correctly', () => {
    const post = {
      id: '1',
      title: 'This is my article',
      intro: 'This is the intro text to my article',
      body: 'This is the **body** of my article, which includes markdown',
      created_at: new Date(Date.UTC(96, 1, 2, 3, 4, 5)),
      comments: [{
        id: '1',
        name: 'Dave Smith',
        body: 'This is the **body** of my comment, which includes markdown',
        created_at: new Date(Date.UTC(97, 1, 2, 3, 4, 5))
      }]
    }

    const component = shallow(
      <Article post={post} />
    )

    expect(component).toMatchSnapshot()
  })
})
