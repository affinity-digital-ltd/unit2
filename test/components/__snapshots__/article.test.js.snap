// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Article /> it renders correctly 1`] = `
<div
  className="row"
>
  <div
    className="col-lg-8 offset-lg-2"
  >
    <h1
      className="c-article__header"
    >
      This is my article
    </h1>
    <hr
      className="c-article__rule"
    />
    <p
      className="c-article__intro"
    >
      This is the intro text to my article
    </p>
    <p
      className="c-article__body"
    >
      This is the **body** of my article, which includes markdown
    </p>
    <div
      className="c-article__comments"
    >
      <div
        className="card c-article__comment"
        key="0"
      >
        <div
          className="card-header"
        >
          <time
            dateTime={1997-02-02T03:04:05.000Z}
          >
            February 2, 1997
          </time>
        </div>
        <div
          className="card-body"
        >
          <h6
            className="card-subtitle mb-2 text-muted"
          >
            Dave Smith
          </h6>
          This is the **body** of my comment, which includes markdown
        </div>
      </div>
    </div>
    <div
      className="card"
    >
      <div
        className="card-body"
      >
        <p>
          If you would like to leave a comment, then please login
        </p>
        <a
          className="btn btn-success btn-lg"
          onClick={[Function]}
        >
          Login
        </a>
      </div>
    </div>
  </div>
</div>
`;
